### PR TITLE
dune 1.11: add conflict with num =1.0

### PR DESCRIPTION
**Pull request by mseri**
This is a test to check if the error
```

#=== ERROR while compiling dune.1.11.0 ========================================#
# context              2.1.1 | linux/x86_64 | ocaml-base-compiler.4.10.2 | file:///home/opam/opam-repository
# path                 ~/.opam/4.10/.opam-switch/build/dune.1.11.0
# command              ~/.opam/opam-init/hooks/sandbox.sh build ./boot.exe --release -j 31
# exit-code            1
# env-file             ~/.opam/log/dune-99-bf8fcb.env
# output-file          ~/.opam/log/dune-99-bf8fcb.out
### output ###
# Error: stat: /home/opam/.opam/4.10/lib/ocaml/arith_flags.cmx: No such file or directory
# -> required by src/stdune/.stdune.objs/native/stdune__Appendable_list.cmx
# -> required by src/catapult/.catapult.objs/native/catapult.cmx
# -> required by src/.dune.objs/native/dune__Action.cmx
# -> required by bin/.main.objs/native/main.cmx
# -> required by bin/main.a
# -> required by bin/main/main_dune.exe
# -> required by install bin/dune
# -> required by dune.install
```
seen on Frama-C PR is due to a problem in num =1.0

Signed-off-by: Marcello Seri <marcello.seri@gmail.com>

### CODE REVIEW

**Comment by dra27** [comment]
The conflict doesn't feel right, and marking num.1.0 as unavailable definitely doesn't (lockfiles, etc.). Reproducing locally, say, https://opam.ci.ocaml.org/github/ocaml/opam-repository/commit/73d98347c3c6abab26619e872fabd5648cca833d/variant/opam-2.1,compilers,4.11,frama-c.24.0,lower-bounds, I don't have any problems installing num.1.0 and dune.1.11.0 (in fact, the build failed because of C compiler problems with lablgtk).

**Comment by dra27** [comment]
I'm not sure what the underlying issue is without being able to reproduce it with an entire log - I was wondering if it's that `num` is only partially installed, but that seems surprising, given the number of instances of it in the original PR.

File: packages/dune/dune.1.11.0/opam [hunk 1]
```diff
@@ -34,6 +34,7 @@ conflicts: [
   "jbuilder" {!= "transition"}
   "odoc" {< "1.3.0"}
   "dune-release" {< "1.3.0"}
+  "num" {= "1.0"}
```
Comment (mseri) [inline packages/dune/dune.1.11.0/opam:37]:
```suggestion
```

Let's make an experiment and see if removing this lets the issue appear

**Comment by mseri** [comment]
> The conflict doesn't feel right, 

I agree, it was just to see if the issue goes away with it.

> and marking num.1.0 as unavailable definitely doesn't (lockfiles, etc.).

Then I'd like to understand why it happens. It was clearly a problem with num, some files where missing

**Comment by mseri** [comment]
Maybe the issue is here:
```
-> installed base-bytes.base
[ERROR] The compilation of dune.1.11.0 failed at "./boot.exe --release -j 31".
-> installed camlzip.1.06
-> installed num.1.0
```

is it possible that it is trying to compile dune before num is installed and some dependencies of dune, or dune itself, actually need num?

**Comment by mseri** [comment]
In that case, dune should depend on num because `stdune` and `catapult` do

**Comment by mseri** [comment]
That's also not it, since without num is working fine. I don't know. I'll close this, it was to check if it was related or not, and I agree that the issue must be hidden somewhere else

**Pull request closed without merging**